grammer.html

```diff
@@ -7,6 +7,7 @@
 <term> ::= <factor> | <term> <mulop> <factor>
 <factor> ::= <number> | '(' <expression> ')' | <addop> <factor> | <identifier> | <string> | <atom>
 <atom> ::= <if-expression> | <while-expression> | <for-expression>
+<function-definition> ::= 'func' <identifier>? '(' (<identifier> [',' <identifier>]*)? ')' '-' '>' <expression>
```

Comment:
Shouldn't out grammar for functions be like `func(<zero or more identifiers seperated by commas>) { <statement> }`? Also the `+ ?` and `()` are usually not found in BNFs right?